src/Inciter/AMR/edge.h

```diff
@@ -0,0 +1,80 @@
+#ifndef AMR_edge_t_h
+#define AMR_edge_t_h
+
+#include <iostream>
+#include <ostream>
+
+class edge_t {
+    using edge_ = std::pair<size_t, size_t>;
+    private:
+        edge_ data;
+        friend std::ostream& operator<<(std::ostream&, const edge_t&);
+
+    public:
+
+        edge_ get_data() const {
+            return data;
+        }
+
+        // Constructors
+        edge_t()
+        {
+        }
+        edge_t(size_t A, size_t B)
+        {
+            data = std::make_pair(std::min(A,B), std::max(A,B));
+        }
+
+        // Copy constructor
+        edge_t(const edge_t &e)
+        {
+            data = e.data;
+        }
+
+        // TODO: Sanity check this
+        edge_t& operator=(const edge_t& e)
+        {
+            data = e.data;
+            return *this;
+        }
```

Comment:
I bet both the copy constructor and assignment can be just deleted and the compiler will auto-generate them.